Print version at startup and export it to disk

lbonaldo opened the pull request:
## Description

This PR adds two new functions to the codebase:
- `print_genx_version()` that prints the version of GenX to the console when loading GenX. 
- `write_system_env_summary()` that writes a summary of the current testing environment to a YAML file. 

This information can be helpful for reproducing and analyzing test results.

## What type of PR is this? (check all applicable)

- [X] Feature

## Related Tickets & Documents
It addresses issue #708.

## Checklist

- [X] Code changes are sufficiently documented; i.e. new functions contain docstrings and .md files under /docs/src have been updated if necessary.
- [X] The latest changes on the target branch have been incorporated, so that any conflicts are taken care of before merging. This can be accomplished either by merging in the target branch (e.g. 'git merge develop') or by rebasing on top of the target branch (e.g. 'git rebase develop'). Please do not hesitate to reach out to the GenX development team if you need help with this.
- [X] Code has been tested to ensure all functionality works as intended.
- [x] CHANGELOG.md has been updated (if this is a 'notable' change).
- [X] I consent to the release of this PR's code under the GNU General Public license.

## How this can be tested

By running the example cases. 

## Post-approval checklist for GenX core developers
After the PR is approved

- [x] Check that the latest changes on the target branch are incorporated, either via merge or rebase
- [x] Remember to squash and merge if incorporating into develop

sambuddhac reviewed (commented):
OOOHHHH !!!!!!! Fancy !!!!!! <3

sambuddhac reviewed (approved):
Minor comment and question:
1. Does the init function work similar way as in Python? Since there isn't any data to initialize in this case, is this always the first function that is executed by design?
2. Is there a way the top couple horizontal lines in the GenX logo in the opening page (as part of the newly defined function) be realigned a bit?

lbonaldo commented:
> Minor comment and question:
> 
>     1. Does the init function work similar way as in Python? Since there isn't any data to initialize in this case, is this always the first function that is executed by design?
> 
>     2. Is there a way the top couple horizontal lines in the GenX logo in the opening page (as part of the newly defined function) be realigned a bit?

1. The `__init__` function is a method that runs automatically when the module is loaded. I moved the `print_genx_version()` function call there to ensure the version is printed after a user imports GenX.
2. Sure, I fixed it.

sambuddhac commented:
> > Minor comment and question:
> > ```
> > 1. Does the init function work similar way as in Python? Since there isn't any data to initialize in this case, is this always the first function that is executed by design?
> > 
> > 2. Is there a way the top couple horizontal lines in the GenX logo in the opening page (as part of the newly defined function) be realigned a bit?
> > ```
> 
> 1. The `__init__` function is a method that runs automatically when the module is loaded. I moved the `print_genx_version()` function call there to ensure the version is printed after a user imports GenX.
> 2. Sure, [I fixed it](https://github.com/GenXProject/GenX.jl/pull/712/commits/7a4b8afb0a81b1b6851137c58203ae9683fc7038).

Perfect !!!! Thanks a lot !!!
